torchao/prototype/quantization/safetensors_support.py

```diff
@@ -0,0 +1,163 @@
+import json
+from typing import Dict, Tuple
+
+import torch
+from safetensors.torch import load_file, save_file
+
+from torchao.float8.inference import Float8MMConfig
+from torchao.prototype.quantization.QuantizeTensorToFloat8KwargsJSON import (
+    QuantizeTensorToFloat8KwargsJSONEncoder,
+    config_from_dict,
+)
+from torchao.quantization import Float8Tensor
+from torchao.quantization.quantize_.workflows.float8.float8_tensor import (
+    QuantizeTensorToFloat8Kwargs,
+)
+
+
+def load_tensor_subclass_dict(file_path: str, device: str):
+    """
+    Load a dictionary of tensor subclasses from a safetensors file.
+
+    Args:
+        file_path: Path to the safetensors file
+
+    Returns:
+        Dictionary of reconstructed tensor subclasses
+    """
+    loaded_tensors = load_file(file_path, device)
+
+    with open(file_path, "rb") as f:
+        import struct
+
+        header_size = struct.unpack("<Q", f.read(8))[0]
+        header_bytes = f.read(header_size)
+        header = json.loads(header_bytes)
+        metadata = header.get("__metadata__", {})
+
+    if "tensor_names" not in metadata:
+        raise ValueError("No tensors found")
+
+    tensor_names = json.loads(metadata["tensor_names"])
+    result = {}
+
+    for tensor_name in tensor_names:
+        tensor_metadata = {}
+        for key, value in metadata.items():
+            if key.startswith(f"{tensor_name}:"):
+                # Remove the prefix
+                tensor_metadata[key[len(tensor_name) + 1 :]] = value
+
+        tensor_tensors = {}
+        for key, value in loaded_tensors.items():
+            if key.startswith(f"{tensor_name}:"):
+                # Remove the prefix
+                tensor_tensors[key[len(tensor_name) + 1 :]] = value
+
+        tensor_type = tensor_metadata.get("tensor_type")
+
+        if tensor_type == str(Float8Tensor):
+            # not the same as mm_config in act_quant_kwargs
+            mm_config = json.loads(tensor_metadata.get("mm_config"))
+            if mm_config:
+                mm_config = Float8MMConfig(*mm_config)
```

Comment:
why do we need to initialize mm_config instead of just reading from whatever is serialized?